src/main/java/com/amazon/aocagent/services/ECSService.java

```diff
@@ -0,0 +1,354 @@
+import com.amazonaws.services.ecs.model.DeleteClusterRequest;
+import com.amazonaws.services.ecs.model.DeregisterTaskDefinitionRequest;
+import com.amazonaws.services.ecs.model.DescribeClustersRequest;
+import com.amazonaws.services.ecs.model.DescribeClustersResult;
+import com.amazonaws.services.ecs.model.DesiredStatus;
+import com.amazonaws.services.ecs.model.KeyValuePair;
+import com.amazonaws.services.ecs.model.LaunchType;
+import com.amazonaws.services.ecs.model.ListTaskDefinitionsResult;
+import com.amazonaws.services.ecs.model.ListTasksRequest;
+import com.amazonaws.services.ecs.model.LogConfiguration;
+import com.amazonaws.services.ecs.model.LogDriver;
+import com.amazonaws.services.ecs.model.NetworkConfiguration;
+import com.amazonaws.services.ecs.model.NetworkMode;
+import com.amazonaws.services.ecs.model.PortMapping;
+import com.amazonaws.services.ecs.model.RegisterTaskDefinitionRequest;
+import com.amazonaws.services.ecs.model.RegisterTaskDefinitionResult;
+import com.amazonaws.services.ecs.model.RunTaskRequest;
+import com.amazonaws.services.ecs.model.RunTaskResult;
+import com.amazonaws.services.ecs.model.StopTaskRequest;
+import com.amazonaws.services.ecs.model.TransportProtocol;
+import com.amazonaws.services.logs.AWSLogs;
+import com.amazonaws.services.logs.AWSLogsAsyncClientBuilder;
+import com.amazonaws.services.logs.model.CreateLogGroupRequest;
+import com.amazonaws.services.logs.model.DescribeLogGroupsRequest;
+import com.amazonaws.services.logs.model.DescribeLogGroupsResult;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+@Log4j2
+public class ECSService {
+  private static final String AWS_LOGS_REGION_KEY = "awslogs-region";
+  private static final String AWS_LOGS_GROUP_KEY = "awslogs-group";
+  private static final String AWS_LOGS_STREAM_PREFIX_KEY = "awslogs-stream-prefix";
+  private static final String ECS_PREFIX = "ecs";
+  private static final String AOC_EMITTER = "aoc-emitter";
+  private static final String EC2 = "EC2";
+  private static final String INSTANCE_ID = "otlp_instance_id";
+  private static final String OTLP_ENDPOINT = "otlp_endpoint";
+  private AmazonECS ecsClient;
+  private AWSLogs awsLogsClient;
+
+  public ECSService(final String region) {
+    this.ecsClient = AmazonECSClientBuilder.standard().withRegion(region).build();
+    this.awsLogsClient = AWSLogsAsyncClientBuilder.standard().withRegion(region).build();
+  }
+
+  /**
+   * create ECS cluster for integ testing.
+   * @return cluster creation result
+   */
+  public CreateClusterResult createCluster() {
+    CreateClusterRequest request =
+        new CreateClusterRequest().withClusterName(GenericConstants.ECS_SIDECAR_CLUSTER.getVal());
+    log.info("creating ECS cluster: {}", GenericConstants.ECS_SIDECAR_CLUSTER.getVal());
```

Comment:
race condition if two ecs tests are running.

Reply:
good point. I can create different cluster in the runtime, but not sure how to clean it because clean and setup are in different processes.

Reply:
one possible way is that you give a timestamp as the suffix of the cluster name, and clean it base on the timestamp like clean the one older than 2 hours.